File: src/main/frontend/src/model/Cargo.ts
export class Cargo {

  public static from(formData: FormData): Cargo {
    return new Cargo(
        formData.get("id") as string | null,
        formData.get("name") as string | null,
        formData.get("proprietor") as string | null,
        formData.get("contactNo") as string | null,
        formData.get("address") as string | null,
        formData.get("reference") as string | null,
    );
  }

  public toObject() {
    return {
      name: this.name,
      proprietor: this.proprietor,
      contactNo: this.contactNo,
      address: this.address,
      reference: this.reference,
    };
  }

  public constructor(
      public id: string | null,
      public name: string | null,
      public proprietor: string | null,
      public contactNo: string | null,
      public address: string | null,
      public reference: string | null,
  ) {
  }
}

File: src/main/java/com/raihanorium/springreact/model/Cargo.java
package com.raihanorium.springreact.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Cargo {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String proprietor;
    private String contactNo;
    private String address;
    private String reference;
}
